### ISSUE TITLE
Upgrade from 4.5.4 to 50-RC3 loses datastores and job lists

### ISSUE THREAD

**Issue by LosD** [datacleaner/DataCleaner#1195]
When upgrading directly from to 4.5.4 to 5.0-RC3, datastores and job lists may be lost. Reference data seems to survive.

This may be related to the change we did recently regarding writing reference data to conf.xml (sorry about the missing issue number, at a customer now)

**Comment by kaspersorensen** [comment]
I think this is either fixed or no longer relevant since 5.0 was released as GA :)

**Comment by LosD** [comment]
Not sure about that. It still hits anyone who upgrades to 5.0.

Of course, it slowly gets less and less relevant ☺

I think at some point we should find a better solution than the .datacleaner/version scheme. At least so far it has mostly been an annoyance.

**Comment by kaspersorensen** [comment]
I agree the scheme hasn't been a big success, but a new scheme should be registered in a separate issue, shouldn't it?

**Comment by LosD** [comment]
Certainly, that was mostly just an aside. I guess we'll introduce an issue with the new scheme... When we know what we want the new scheme to be :)

I'd like to leave this one open for now. It pains me that we are annoying users for no benefit.

**Comment by kaspersorensen** [comment]
But... This issue here only happened because of the -RCn suffix I think. I believe that an upgrade to any properly release version works fine. Or no?

**Comment by LosD** [comment]
I'd have to investigate, but I think we originally made it a rule that major versions wouldn't get a copy, as we thought incompatibilities were almost a given.

But it's so long ago that I could certainly be wrong.
